presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -341,11 +341,7 @@ private PartitionStatistics loadPartitionColumnStatistics(WithIdentity<HiveParti
                 partition.getIdentity(),
                 table,
                 ImmutableList.of(getExistingPartition(partition.getIdentity(), table, partition.getKey().getPartitionValues())));
-        PartitionStatistics result = partitionStatistics.get(partitionName);
-        if (result == null) {
-            throw new PrestoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Statistics result does not contain entry for partition: " + partition.getKey().getPartitionName());
```

Comment:
What's redundant about this check?

Reply:
Once you have existing partition, then `getPartitionStatistics` cannot fail with this error (at least basic stats are present),